---
title: "Lab 8 Homework"
author: "Vincent Castillo"
date: "`r Sys.Date()`"
output:
  html_document:
    keep_md: yes
    theme: darkly
    toc: no
    toc_float: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Instructions
Answer the following questions and complete the exercises in RMarkdown. Please embed all of your code and push your final work to your repository. Your final lab report should be organized, clean, and run free from errors. Remember, you must remove the `#` for any included code chunks to run.  

## Libraries


```{r}
library(ape)
library(seqinr)
library(rentrez)
```


**1. All the databases associated with NCBI are updating and sharing their data daily. With the new strain of the coronavirus, SARS-CoV-2 at the forefront of a lot of medical research, genomes have already been uploaded to GenBank. The accession number for a complete genome of an isolate from China is "MT093631.1". Pull this sequence from GenBank into R, then write it as a .fasta file and read the .fasta file into R using the code we learned in class.**


```{r}
covid19seq <- read.GenBank("MT093631.1", as.character = TRUE)
write.fasta(names = "COV-19", sequences = covid19seq, file.out="data/cov19.fasta")
covid19seq2 <- read.fasta(file = "data/cov19.fasta")
covid19seq3 <- covid19seq2[[1]]
```


**2. Explore this sequence using functions we learned in class. What is the length of the genome? Create a table showing the nucleotide composition of the sequence, and calculate the GC content.**


```{r}
length(covid19seq3)
table(covid19seq3)
GC(covid19seq3)
```


**3.Several genomes for the new strain of coronavirus have already been uploaded to GenBank. Using search terms, look for uploaded genomes SARS-CoV-2. (Hint: You're looking for things listed as genomes, which you won't find looking for a [GENE] keyword 'entrez_db_searchable("nuccore")' will provide you with a list of search term options.) Limit your returned hits to 10. Download your ten genomes and write them out as a .fasta file.**


```{r}
entrez_db_searchable("nuccore")
cov2 <- entrez_search(db = "nuccore", term= "SARS-CoV-2 [ORGN]", retmax = 10)
cov2seqs <- entrez_fetch(db = "nuccore", id = cov2$ids, rettype = "fasta")
cov2seqs
write(cov2seqs, "data/cov2_fasta.fasta", sep = "\n")
```

